workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/ComposedViewFactory.kt

```diff
@@ -0,0 +1,141 @@
+// See https://youtrack.jetbrains.com/issue/KT-31734
+@file:Suppress("RemoveEmptyParenthesesFromAnnotationEntry")
+
+package com.squareup.workflow1.ui.compose
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.ComposeView
+import com.squareup.workflow1.ui.LayoutRunner
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.ViewFactory
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.bindShowRendering
+import kotlin.reflect.KClass
+
+/**
+ * Creates a [ViewFactory] that uses a [Composable] function to display the rendering.
+ *
+ * Simple usage:
+ *
+ * ```
+ * val FooViewFactory = composedViewFactory { rendering, _ ->
+ *   Text(rendering.message)
+ * }
+ *
+ * …
+ *
+ * val viewRegistry = ViewRegistry(FooViewFactory, …)
+ * ```
+ *
+ * If you need to write a class instead of a function, for example to support dependency injection,
+ * see [ComposedViewFactory].
+ *
+ * For more details about how to write composable view factories, see [ComposedViewFactory].
+ */
+@WorkflowUiExperimentalApi
+public inline fun <reified RenderingT : Any> composedViewFactory(
+  noinline content: @Composable (
+    rendering: RenderingT,
+    environment: ViewEnvironment
+  ) -> Unit
+): ViewFactory<RenderingT> = composedViewFactory(RenderingT::class, content)
+
+@PublishedApi
+@WorkflowUiExperimentalApi
+internal fun <RenderingT : Any> composedViewFactory(
+  type: KClass<RenderingT>,
+  content: @Composable (
+    rendering: RenderingT,
+    environment: ViewEnvironment
+  ) -> Unit
+): ViewFactory<RenderingT> = object : ComposedViewFactory<RenderingT>() {
+  override val type: KClass<in RenderingT> = type
+  @Composable override fun Content(
+    rendering: RenderingT,
+    viewEnvironment: ViewEnvironment
+  ) {
+    content(rendering, viewEnvironment)
+  }
+}
+
+/**
+ * A [ViewFactory] that uses a [Composable] function to display the rendering. It is the composable
+ * version of [LayoutRunner].
+ *
+ * Simple usage:
+ *
+ * ```
+ * class FooViewFactory : ComposedViewFactory<Foo>() {
+ *   override val type = Foo::class
+ *
+ *   @Composable override fun Content(
+ *     rendering: Foo,
+ *     viewEnvironment: ViewEnvironment
+ *   ) {
+ *     Text(rendering.message)
+ *   }
+ * }
+ *
+ * …
+ *
+ * val viewRegistry = ViewRegistry(FooBinding, …)
+ * ```
+ *
+ * ## Nesting child renderings
+ *
+ * Workflows can render other workflows, and renderings from one workflow can contain renderings
+ * from other workflows. These renderings may all be bound to their own [ViewFactory]s. Regular
+ * [ViewFactory]s and `LayoutRunner`s use
```

Comment:
nit: \`LayoutRunner\` ->  [LayoutRunner]? (We use the brackets with LayoutRunner above, at any rate.)